### PR TITLE
Add error check and improve msg

### PR DESCRIPTION
**Pull request by yeasy**
The patchset checks the error when doing yaml marshal, and improves the message when error happens.

Change-Id: I40fb7dbe930bcdc6563078daab50f4f4564c3381

#### Type of change

- Improvement (improvement to code, performance, etc)

#### Description

* Check the possible error in marshaling;
* Improve message when error happens.

#### Additional details

N/A

#### Related issues

N/A

#### Release Note

N/A

### CODE REVIEW

**Comment by denyeart** [comment]
Well, your improved message has proven effective... this test needs to be updated to match the improved message:

```
--- FAIL: TestMissingArguments (35.34s)
    main_test.go:30: 
        Got stuck at:
            Failed persisting configuration: config isn't valid: IdentityPath is mandatory and cannot be empty
            
        Waiting for:
            empty string that is mandatory
FAIL
	github.com/hyperledger/fabric/cmd/discover	coverage: 0.0% of statements
```

**Comment by yeasy** [comment]
> IdentityPath is mandatory and cannot be empty

Aha, just updated!

Could you help rerun the CI test? @denyeart

File: cmd/discover/main_test.go [hunk 1]
```diff
@@ -27,5 +27,5 @@ func TestMissingArguments(t *testing.T) {
 	process, err := Start(cmd, nil, nil)
 	gt.Expect(err).NotTo(HaveOccurred())
 	gt.Eventually(process, 5*time.Second).Should(Exit(1))
-	gt.Expect(process.Err).To(gbytes.Say("empty string that is mandatory"))
+	gt.Expect(process.Err).To(gbytes.Say("IdentityPath is mandatory and cannot be empty"))
```
Comment (satota2) [inline cmd/discover/main_test.go]: Hi @yeasy
In the Go v1 series, when looping through a map with a range, the order in which items are retrieved is randomized by design. 
https://go.dev/doc/go1#iteration

Therefore, it is necessary to update the expected values for the tests accordingly to accommodate both cases, or make modifications to identify the element (e.g., setting only one missing flag).
```suggestion
	gt.Expect(process.Err).To(gbytes.Say("IdentityPath|KeyPath is mandatory and cannot be empty"))
```

Comment (yeasy) [reply cmd/discover/main_test.go]: Updated, thanks!

**Review by denyeart — APPROVED** [review]
Thanks @yeasy

**Pull request merged**